pkg/metrics/registry.go

```diff
@@ -232,7 +232,7 @@ func (r *Registry) DumpMetrics() ([]*models.Metric, error) {
 			case dto.MetricType_SUMMARY:
 				value = metricLabel.GetSummary().GetSampleSum()
 			case dto.MetricType_HISTOGRAM:
-				value = metricLabel.GetHistogram().GetSampleSum()
+				value = float64(metricLabel.GetHistogram().GetSampleCount())
```

Comment:
Also, nit, there appears to be a metricLabel.GetSampleCountFloat function to cast to f64.

Reply:
1) tbh this change is not necessary for this PR in the current state.
Initially, I was trying to observe values of infinity in the histogram and as the sum hit infinity, the JSON unmarshal for the value was failing. I ultimately decided to observe the TIMEOUT value instead. However, I was thinking that (a) this change could be helpful if infinity values are observed by a histogram metric in the future (I don't think this is entirely impossible) and (b) this function is returning a summary statistic for metrics, and it makes a bit more sense (to me) to summarize a histogram with a Count of observations rather than the Sum of them.
I do see the perspective that this could be a premature change, though, in this case.

2) Based on my understanding, this function is only called for `cilium metrics` dumps. I don't think I see a way that this breaks metrics dumps, as if there is a histogram metric, it must have a count. Do you have any other situations in mind that you could point me towards, I could test them out?

Reply:
That seems reasonable to me :) I will revert the change for this PR as it seems unnecessary here. Thank you for the information, I will keep it in mind!